signer/src/proto/convert.rs

```diff
@@ -1131,8 +1150,22 @@ impl TryFrom<proto::WstsMessage> for WstsMessage {
                 wsts::net::Message::SignatureShareResponse(inner.try_into()?)
             }
         };
+
+        #[allow(deprecated)]
         Ok(WstsMessage {
-            txid: BitcoinTxId::try_from(value.txid.required()?)?.into(),
+            id: match value.id {
+                Some(id) => match id {
+                    wsts_message::Id::IdBitcoinTxid(txid) => {
+                        WstsMessageId::BitcoinTxid(BitcoinTxId::try_from(txid)?.into())
+                    }
+                    wsts_message::Id::IdRotateKey(pubkey) => {
+                        WstsMessageId::RotateKey(PublicKey::try_from(pubkey)?)
+                    }
+                },
+                None => WstsMessageId::BitcoinTxid(
+                    BitcoinTxId::try_from(value.txid.required()?)?.into(),
+                ),
```

Comment:
This is also for backwards compatibility. When converting from a `proto::WstsMessage` to a `message:WstsMessage`, if the incoming `id` is `None` then it will try to revert to using the `txid` as a `BitcoinTxId`.

This can be removed once #1285 is merged.